Lib/_pyrepl/readline.py

```diff
@@ -161,6 +169,11 @@ def get_completions(self, stem: str) -> list[str]:
             result.sort()
         return result
 
+    def get_module_completions(self) -> list[str]:
+        completer = ModuleCompleter(namespace={'__package__': '_pyrepl'})  # TODO: namespace?
```

Comment:
Is this creating one of this guys per request? Should we cache the `ModuleCompleter`?

Reply:
it is, and yes we should cache it. Updated in [5c11124](https://github.com/python/cpython/pull/129329/commits/5c11124c39bbac4e9ec1a1de4b29b7ce445abe5c) (I also moved the completer to a separate file to avoid having to reorder everything when I added it to the `ReadlineConfig`)

Reply:
Also changed in [f4e290a](https://github.com/python/cpython/pull/129329/commits/f4e290a03f1f8e42f47175e372248c62b14b726d) so that each Reader gets a new instance (with its own module cache)